### DISCUSSION TITLE
How to config launch.json to debug source or test code in VSCode

### DISCUSSION THREAD

**Discussion by defpis** [pixijs/pixijs#7558] [Q&A]
Breakpoint in source code or test code is not working.

![image](https://user-images.githubusercontent.com/48018016/123648407-816d9c80-d85b-11eb-8fbf-1a830cd70328.png)

**Comment by miltoncandelero** [comment]
you will need the[ chrome debugger ](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome)but it should be something like this:

```json
{
	"version": "0.2.0",
	"configurations": [
		{
			"type": "chrome",
			"request": "launch",
			"name": "Launch Chrome against localhost",
			"url": "http://localhost:3000",
			"webRoot": "${workspaceFolder}"
		}
	]
}
```

check your port (I use 3000 you might use another)

**Reply by miltoncandelero** [reply]
If you want to use [the firefox debugger](https://marketplace.visualstudio.com/items?itemName=firefox-devtools.vscode-firefox-debug)

```json
	"type": "firefox",
	"name": "Launch Firefox against localhost",
	"request": "launch",
	"reAttach": true,
	"url": "http://localhost:3000",
	"webRoot": "${workspaceFolder}",
```

**Reply by defpis** [reply]
It still doesn't work, can give a debugging example, thanks a million!

**Reply by miltoncandelero** [reply]
what kind of "doesn't work" are we talking here? 🤔 

Does chrome open? does your app/game play inside that chrome?
